src/app/search/search.component.ts

```diff
@@ -60,26 +60,34 @@ export class SearchComponent implements OnInit {
   private searchTerms = new Subject<string>();
   checkedSearchOption: string;
 
-  constructor(private http: Http, private searchService: SearchService) {
-    this.searchFormGroup = new FormGroup({
-      searchOptionCtrl: new FormControl(),
-      queryCtrl: new FormControl()
-    });
+  constructor(
+    readonly http: Http,
```

Comment:
Why are constructor's parameters changed to `readonly`?

Reply:
I want the `http` member to be immutable, as it is being imported from the outside module.  From what I've heard this is a good practice to perform, plus there is a TS docs note about `readonly`: https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#read-only-properties-and-index-signatures